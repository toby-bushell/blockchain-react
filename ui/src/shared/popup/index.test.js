import React from 'react';
import renderer from 'react-test-renderer';
import Popup from './index';
import { shallow, mount, render } from 'enzyme';
import 'jest-styled-components';

// Popup component uses a render array. Values are passed in a render prop and is required

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Popup
        render={() => {
          <p>Firing</p>;
        }}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it('if close button clicked updates state and hides', () => {
  const wrapper = shallow(
    <Popup
      render={close => (
        <div>
          <p>Firing</p>
          <button onClick={close}>Close</button>
        </div>
      )}
    />
  );
  wrapper.find('button').simulate('click');

  expect(wrapper.state('active')).toEqual(false);
  expect(wrapper).toHaveStyleRule('display', 'none');
});
