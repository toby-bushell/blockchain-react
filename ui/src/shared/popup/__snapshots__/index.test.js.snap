// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  width: 400px;
  max-width: calc(100% - 40px);
  padding: 40px;
  text-align: center;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
</div>
`;
